tests/optim/test_optim.py

```diff
@@ -75,6 +75,38 @@ def optim_params(module_name, param_name):
         assert fixed_param_unchanged and not free_param_unchanged
 
 
+@pytest.mark.parametrize('scheduler', [optim.LambdaLR({'optimizer': torch.optim.SGD, 'optim_args': {'lr': 0.01},
+                                                       'lr_lambda': lambda epoch: 2. * epoch}),
+                                       optim.StepLR({'optimizer': torch.optim.SGD, 'optim_args': {'lr': 0.01},
+                                                     'gamma': 2, 'step_size': 1}),
+                                       optim.ExponentialLR({'optimizer': torch.optim.SGD, 'optim_args': {'lr': 0.01},
+                                                            'gamma': 2})])
+def test_dynamic_lr(scheduler):
+    pyro.clear_param_store()
+
+    def model():
+        sample = pyro.sample('latent', Normal(torch.tensor(0.), torch.tensor(0.3)))
+        return pyro.sample('obs', Normal(sample, torch.tensor(0.2)), obs=torch.tensor(0.1))
+
+    def guide():
+        loc = pyro.param('loc', torch.tensor(0.))
+        scale = pyro.param('scale', torch.tensor(0.5))
+        pyro.sample('latent', Normal(loc, scale))
+
+    svi = SVI(model, guide, scheduler, loss=TraceGraph_ELBO())
+    for epoch in range(2):
+        svi.step()
```

Comment:
Can you add a test with multiple steps per epoch?

Reply:
The PyTorch lr_scheduler interface has an optional `epoch` argument that would be used like
```py
for epoch in range(10):
    for minibatch in batcher(data):
        compute_loss(minibatch).backward()
        optim.step(epoch=epoch)
```
You'll need more plumbing to use the `epoch` kwarg in Pyro, something like
```py
optim = LambdaLR(...)
svi = SVI(model, guide, optim, Trace_ELBO())
for epoch in range(10):
    for minibatch in batcher(data):
        optim.set_epoch(epoch)  # inform the scheduler of the latest epoch
        svi.step()
```

Reply:
on `step` - yeah i plumbed that through but `last_epoch` behaves weirdly. it should use the index of the epoch you specify. let me investigate it a bit i may be using it wrong